projects/every-voice/src/lib/every-voice/every-voice.component.ts

```diff
@@ -1,37 +1,77 @@
-import { ChangeDetectorRef, Component, OnInit, Input } from "@angular/core";
+import {
+  ChangeDetectorRef,
+  Component,
+  OnInit,
+  Input,
+  OnDestroy,
+} from "@angular/core";
 import { EveryVoiceService } from "../every-voice.service";
+import { BehaviorSubject, Subject, Subscription } from "rxjs";
 
 @Component({
   selector: "lib-every-voice",
   templateUrl: "./every-voice.component.html",
   styleUrls: ["./every-voice.component.css"],
 })
-export class EveryVoiceComponent implements OnInit {
+export class EveryVoiceComponent implements OnInit, OnDestroy {
   @Input() textToGenerate: string;
   @Input() icon = "volume_up";
   @Input() stopIcon = "stop";
-  isPlaying = false;
-  constructor(public tts: EveryVoiceService, private cdr: ChangeDetectorRef) {}
+  isPlaying: boolean;
+  isLoading: boolean;
+  hasError: boolean;
 
-  ngOnInit(): void {}
+  private subscription: Subscription;
+
+  constructor(public tts: EveryVoiceService, private cdr: ChangeDetectorRef) {
+    this.isLoading = false;
+    this.subscription = new Subscription();
```

Comment:
nice, yes I guess this is basically the same as the unsubscribe$ = Subject<void> & takeUntil(this.unsubscribe) pattern.